playbooks/cloud-provider-openstack-acceptance-test-flexvolume-cinder/run.yaml

```diff
@@ -0,0 +1,145 @@
+          # https://github.com/liu-sheng/cloud-provider-openstack/pull/17
+          cp -f pkg/flexvolume/cinder_client.go vendor/k8s.io/cloud-provider-openstack/pkg/flexvolume/cinder_client.go
+          # cp -f pkg/flexvolume/cinder_baremetal_util.go vendor/k8s.io/cloud-provider-openstack/pkg/flexvolume/cinder_baremetal_util.go
+          cd /home/zuul/src/k8s.io/cloud-provider-openstack && CGO_ENABLED=0 GOOS=linux go build \
+              -ldflags "-X 'main.version=19c9e833-dirty'" \
+              -o cinder-flex-volume-driver \
+              cmd/cinder-flex-volume-driver/main.go
+          cd -
+
+          mkdir -p /usr/libexec/kubernetes/kubelet-plugins/volume/exec/cinder.io~cinder-flex-volume-driver/
+          cp cinder-flex-volume-driver /usr/libexec/kubernetes/kubelet-plugins/volume/exec/cinder.io~cinder-flex-volume-driver/
+
+          export k8s_root="$GOPATH/src/k8s.io/kubernetes"
+          export kubectl="$k8s_root/cluster/kubectl.sh"
+          # -E preserves the current env vars, but we need to special case PATH
+          sudo -E PATH=$PATH SHELLOPTS=$SHELLOPTS "$k8s_root/hack/local-up-cluster.sh" -O
+
+          # set up the config we need for kubectl to work
+          "$kubectl" config set-cluster local --server=https://localhost:6443 --certificate-authority=/var/run/kubernetes/server-ca.crt
+          "$kubectl" config set-credentials myself --client-key=/var/run/kubernetes/client-admin.key --client-certificate=/var/run/kubernetes/client-admin.crt
+          "$kubectl" config set-context local --cluster=local --user=myself
+          "$kubectl" config use-context local
+
+          # Hack for RBAC for all for the new cloud-controller process, we need to do better than this
+          "$kubectl" create clusterrolebinding --user system:serviceaccount:kube-system:default kube-system-cluster-admin-1 --clusterrole cluster-admin
+          "$kubectl" create clusterrolebinding --user system:serviceaccount:kube-system:pvl-controller kube-system-cluster-admin-2 --clusterrole cluster-admin
+          "$kubectl" create clusterrolebinding --user system:serviceaccount:kube-system:cloud-node-controller kube-system-cluster-admin-3 --clusterrole cluster-admin
+          "$kubectl" create clusterrolebinding --user system:serviceaccount:kube-system:cloud-controller-manager kube-system-cluster-admin-4 --clusterrole cluster-admin
+          "$kubectl" create clusterrolebinding --user system:serviceaccount:kube-system:shared-informers kube-system-cluster-admin-5 --clusterrole cluster-admin
+          "$kubectl" create clusterrolebinding --user system:kube-controller-manager  kube-system-cluster-admin-6 --clusterrole cluster-admin
+
+          # Avoid missing feature 400000000000000 error in ceph
+          # http://lists.ceph.com/pipermail/ceph-users-ceph.com/2017-July/019383.html
+          ceph osd crush tunables hammer
+
+          # Run test
+          VOLUME_ID=$(cinder create 1 | awk '/ id / {print $4}')
```

Comment:
Actually I used openstakcclient at start but failed because it's a cinder standalone environment with only cinder and keystone services installed, only the cinderclient works properly.

Reply:
My fault, forget that limit, thank you for clarifying.